pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -178,13 +179,57 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	err = c.Watch(
 		&source.Kind{Type: &corev1.Secret{}},
 		&handler.EnqueueRequestsFromMapFunc{
-			ToRequests: mapFn,
+			ToRequests: targetCredSecretMapFunc,
 		},
 		p)
 	if err != nil {
 		return err
 	}
 
+	// allCredRequestsMapFn simply looks up all CredentialsRequests and requests they be reconciled.
+	allCredRequestsMapFn := handler.ToRequestsFunc(
+		func(a handler.MapObject) []reconcile.Request {
+			log.Info("requeueing all CredentialsRequests")
+			crs := &minterv1.CredentialsRequestList{}
+			err := mgr.GetClient().List(context.TODO(), crs)
+			var requests []reconcile.Request
+			if err != nil {
+				log.WithError(err).Error("error listing all cred requests for requeue")
```

Comment:
This means that we never force the reconcile of all the CredentialsRequests?
Should we exit(1) here to force the Pod to restart?

Reply:
If the admin secret or config map changed yes, we would not reconcile them all, but that is the state we've been in for several releases. Exiting the pod seems like it could be bad precedent, I'm not sure what should be done in a map func when there's an error.

Reply:
Yeah, the inability to return an error really complicates things here. We're probably okay as things will eventually settle to the desired state as the objects get re-reconciled on whatever schedule we have it set to.